src/components/segmentedControl/segmentedControl.svelte

```diff
@@ -0,0 +1,222 @@
+<script context="module" lang="ts">
+  declare global {
+    namespace JSX {
+      interface IntrinsicElements {
+        'leo-segmented-item': HTMLAttributes<HTMLElement> & {
+          // Note: This should line up with Reacts key type, but we don't want
+          // to depend on React in this layer, so we just define it manually.
+          key?: string | number | null
+
+          value?: string
+          children?: any
+        }
+      }
+    }
+  }
+</script>
+
+<script lang="ts">
+  import type { HTMLAttributes } from 'svelte/elements'
+  import { createEventDispatcher, onMount } from 'svelte'
+
+  type Size = 'default' | 'small' | 'tiny'
+
+  export let selected: string | undefined = undefined
```

Comment:
Optional: We could match the native controls and use `value` instead of `selected` here.